### PR TITLE
[location-component] Add OnIndicatorPositionChangedListener and OnIndicatorBearingChangedListener.

### PR DESCRIPTION
**Pull request by pengdev**
PRs must be submitted under the terms of our Contributor License Agreement [CLA](https://github.com/mapbox/mapbox-maps-android/blob/main/CONTRIBUTING.md#contributor-license-agreement).

## Pull request checklist:
 - [x] Briefly describe the changes in this PR.
 - [ ] Include before/after visuals or gifs if this PR includes visual changes.
 - [ ] Write tests for all new functionality. If tests were not written, please explain why.
 - [x] Add example if relevant.
 - [x] Document any changes to public APIs.
 - [x] Apply changelog label ('breaking change', 'bug :beetle:', 'build', 'docs', 'feature :green_apple:', 'performance :zap:', 'testing :100:') or use the label 'skip changelog'
 - [x] Add an entry inside this element for inclusion in the `mapbox-maps-android` changelog: `<changelog>Add OnIndicatorPositionChangedListener and OnIndicatorBearingChangedListener.</changelog>`.

### Summary of changes

This PR adds the OnIndicatorPositionChangedListener and OnIndicatorBearingChangedListener to the location component plugin, these two callbacks will be called on each position and bearing change of the location indicator, including each animation frame.

This PR also updates the LocationComponentActivity to utilise the OnIndicatorPositionChangedListener and always have the location indicator centered.  And LocationTrackingActivity is updated to use LocationComponentPlugin instead of deprecated LocationPlugin.

### CODE REVIEW

File: app/src/main/java/com/mapbox/maps/testapp/examples/LocationTrackingActivity.kt [hunk 1]
```diff
@@ -3,77 +3,115 @@ package com.mapbox.maps.testapp.examples
...
+  private fun setupGesturesListener() {
+    mapView.getGesturesPlugin().addOnMoveListener(onMoveListener)
   }
 
-  override fun onCameraTrackingDismissed() {
-    Toast.makeText(this, "onCameraTrackingDismissed", Toast.LENGTH_SHORT).show()
+  private fun initLocationComponent() {
+    val locationComponentPlugin = mapView.getLocationComponentPlugin()
+    locationComponentPlugin.updateSettings {
+      this.enabled = true
```
Comment (kiryldz) [inline app/src/main/java/com/mapbox/maps/testapp/examples/LocationTrackingActivity.kt:80]: So do I get it correctly that if we miss that line - nothing will work? 😄

Comment (pengdev) [reply app/src/main/java/com/mapbox/maps/testapp/examples/LocationTrackingActivity.kt:80]: right, we have to enable the location component plugin, we can either do it here in the code, or through adding the xml attribute.

**Review by kiryldz — APPROVED** [review]
@pengdev - great job!
As discussed personally - I will add tests later for that new functionality in separate PR.

File: app/src/main/java/com/mapbox/maps/testapp/examples/LocationTrackingActivity.kt (hunk 1 above)
Comment (kiryldz) [reply app/src/main/java/com/mapbox/maps/testapp/examples/LocationTrackingActivity.kt:80]: Can we try enabling plugin (if it's not enabled) if any function (like `updateSettings` in this case) invoked?

Comment (pengdev) [reply app/src/main/java/com/mapbox/maps/testapp/examples/LocationTrackingActivity.kt:80]: I think the setting properties themselves are pretty clear, we should avoid adding too much logic and confuse users.

**Review by Chaoba — APPROVED** [review]

**Pull request merged**